ogr/parsing.py

```diff
@@ -92,22 +92,32 @@ def _prepare_url(potential_url: str) -> Optional[ParseResult]:
         if not potential_url.startswith(
             ("http://", "https://", "git://", "git+https://")
         ):
-            potential_url = "http://" + potential_url
+            potential_url = "https://" + potential_url
 
         return urlparse(potential_url)
 
     def _set_hostname_and_scheme(self, parsed_url: ParseResult):
         self.hostname = parsed_url.hostname
         self.scheme = parsed_url.scheme
 
-    def _parse_username(self, parsed: ParseResult) -> None:
+    def _parse_username(self, parsed: ParseResult) -> bool:
         if ":" not in parsed.netloc:
-            return
+            return True
 
-        # e.g. domain.com:foo or domain.com:1234, where foo is username, but 1234 is port number
         split = parsed.netloc.split(":")
-        if split[1] and not split[1].isnumeric():
-            self.username = split[1]
+        if len(split) > 2:
+            # in case there is port:namespace
+            return False
+
+        if split[1]:
+            if split[1] == "forks":
+                self.is_fork = True
+            elif not split[1].isnumeric():
+                # e.g. domain.com:foo or domain.com:1234,
+                # where foo is username, but 1234 is port number
+                self.username = split[1]
```

Comment:
```suggestion
        if not split[1]:
            return True

        if split[1] == "forks":
            self.is_fork = True
        elif not split[1].isnumeric():
            # e.g. domain.com:foo or domain.com:1234,
            # where foo is username, but 1234 is port number
            self.username = split[1]
```

Maybe?

Reply:
Feel free to ignore..;)

Reply:
Fixed, still gotta do something about this on Friday